### ISSUE TITLE
adb devices showing unauthorized (though no prompt in Android)

### ISSUE THREAD

**Issue by steadfasterX** [Carbon-Fusion/build_fwul#56]
### Description

> openssl-1.1.0.f-2 (containing libcrypto.so.1.1)
> android-tools 8.0.0_r4-1

when first starting adb it will give an error like this:

> List of devices attached
> * daemon not running. starting it now at tcp:5037 *
> ADB server didn't ACK
> * failed to start daemon *
> error: cannot connect to daemon

journal telling this:

```
> Sep 18 14:45:45 fwul kernel: adb[1164]: segfault at 8 ip 00007fcf6a3974b3 sp 00007ffecb357eb0 error 4 in libcrypto.so.1.1[7fcf6a2f3000+256000]
> Sep 18 14:45:45 fwul systemd[1]: Started Process Core Dump (PID 1169/UID 0).
> -- Subject: Unit systemd-coredump@4-1169-0.service has finished start-up
> -- Defined-By: systemd
> -- Support: https://lists.freedesktop.org/55da1e40655a n/a (adb)
>                                              #6  0x00007fcf694a34ca __libc_start_main (libc.so.6)
>                                              #7  0x000055da1e3d660a n/a (adb)
>                                              
>                                              Stack trace of thread 1165:
>                                              #0  0x00007fcf69566e9d poll (libc.so.6)
>                                              #1  0x00007fcf6a0e9d31 n/a (libusb-1.0.so.0)
>                                              #2  0x00007fcf6a77a049 start_thread (libpthread.so.0)
>                                              #3  0x00007fcf69570f0f __clone (libc.so.6)
>                                              
>                                              Stack trace of thread 1166:
>                                              #0  0x00007fcf69566e9d poll (libc.so.6)
>                                              #1  0x00007fcf6a0e36ec n/a (libusb-1.0.so.0)
>                                              #2  0x00007fcf6a0e476a libusb_handle_events_timeout_completed (libusb-1.0.so.0)
>                                              #3  0x00007fcf6a0e485f libusb_handle_events (libusb-1.0.so.0)
>                                              #4  0x000055da1e4211f7 n/a (adb)
>                                              #5  0x00007fcf69e0e00f execute_native_thread_routine (libstdc++.so.6)
>                                              #6  0x00007fcf6a77a049 start_thread (libpthread.so.0)
>                                              #7  0x00007fcf69570f0f __clone (libc.so.6)
>                                              
>                                              Stack trace of thread 1167:
>                                              #0  0x00007fcf6a783d8d __nanosleep (libpthread.so.0)
>                                              #1  0x000055da1e41f85b n/a (adb)
>                                              #2  0x00007fcf69e0e00f execute_native_thread_routine (libstdc++.so.6)
>                                              #3  0x00007fcf6a77a049 start_thread (libpthread.so.0)
>                                              #4  0x00007fcf69570f0f __clone (libc.so.6)
>                                              
>                                              Stack trace of thread 1168:
>                                              #0  0x00007fcf6a7801ad pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
>                                              #1  0x00007fcf69e07ccc __gthread_cond_wait (libstdc++.so.6)
>                                              #2  0x000055da1e3f8d4f n/a (adb)
>                                              #3  0x000055da1e3fa2eb n/a (adb)
>                                              #4  0x00007fcf6a77a049 start_thread (libpthread.so.0)
>                                              #5  0x00007fcf69570f0f __clone (libc.so.6)
> -- Subject: Process 1164 (adb) dumped core
> -- Defined-By: systemd
> -- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
> -- Documentation: man:core(5)
> -- 
> -- Process 1164 (adb) crashed and dumped core.
> -- 
> -- This usually indicates a programming error in the crashing program and
> -- should be reported to its vendor as a bug.
```

1) it makes no difference to use sudo or not.
2) a file `~/.android/adbkey` gets generated but it is ZERO size
3) This happens on several devices of different vendors running MM and N (not sure about O) but as this is not related to the device it doesnt matters much (see the possible solution why)

### Links

- External reports: 
https://forum.xda-developers.com/showpost.php?p=73838714&postcount=405
https://forum.xda-developers.com/android/software-hacking/live-iso-adb-fastboot-driver-issues-t3526755/post73838714
https://forum.xda-developers.com/showpost.php?p=73830526&postcount=397

### Possible solution

Confirmed to fix the problem:

```
adb kill-server
rm ~/.android/adbkey*
sudo pacman -S downgrade
sudo downgrade android-tools-7.1.2_r27-1
```

**Comment by steadfasterX** [comment]
hotfix for FWUL LivePatcher: https://github.com/steadfasterX/arch_fwulpatch/releases/tag/2.2.1

**Comment by steadfasterX** [comment]
Upstream bug report added: https://bugs.archlinux.org/task/55693

**Comment by Scimmia22** [comment]
I think you're confused about the term "upstream". Arch definitely is not upstream here.

**Comment by steadfasterX** [comment]
for FWUL Arch is upstream.

**Comment by Scimmia22** [comment]
Wait, so you're on some kind of derivative distro? <goes off to close Arch ticket>

**Comment by steadfasterX** [comment]
FWUL is an arch based live iso

**Comment by steadfasterX** [comment]
should be fixed in 8.0.0r4-2
